packages/api-queries/README.md

```diff
@@ -10,4 +10,104 @@ Each file represents a resource from `@automattic/api-core`, and exports a query
 
 ## Contributing
 
-To add a new resource, create a new file in the `src` directory and follow the same pattern as the other files in the directory.
+These guidelines should be followed to ensure reusability of the queries and mutations across components.
+
+### File structure
+
+- Add related queries and mutations in the same file inside `src/`.
+  - In most cases, you don't need to group these files into subdirectories.
+- The filename should generally match the directory name of the corresponding resource in `@automattic/api-core`.
+- If a file calls multiple `@automattic/api-core` endpoints, name it according to the feature rather than a single endpoint.
+  - For example, the `site-backup-restore.ts` file calls endpoints under `/sites/${ siteId }/rewind` and `/activity-log/${ siteId }/rewind`.
+
+### Adding queries
+
+- Define queries as functions that return the result of `queryOptions()` from `@tanstack/react-query`. This is to improve TypeScript type inference.
+
+- Avoid using the `select` option directly in the query definition, as it won't be applied when preloading via `queryClient.ensureQueryData()`. Instead:
+  - If the query intends to return only a subset of the data, transform the result in the `queryFn`:
+
+    ```ts
+    export const isSiteUsingBlockThemeQuery = ( siteId: number ) => queryOptions( {
+        ...,
+        queryFn: async () => {
+            const themes = await siteActiveThemesQuery( siteId );
+            return themes[0]?.is_block_theme ?? false;
+        },
+    } );
+    ```
+
+  - If a component only needs a subset of the data, pass `select` from the component:
+
+    ```ts
+    const { data } = useQuery( {
+        ...sitePurchasesQuery( site.ID ),
+        select: ( data ) => data.find( ( purchase ) => purchase.product_slug === WPCOM_DIFM_LITE ),
+    } );
+    ```
+
+- Don't include query options that are specific to a single use case. Add them in the component instead.
+  - For example, don't add an `enabled` option to to the query itself; the component should pass that instead:
+
+    ```ts
+    const { data } = useQuery( {
+        ...siteWordPressVersionQuery( site.ID ),
+        enabled: site.is_wpcom_staging_site,
+    } );
+    ```
+
+- Don't add query options that are specific to a particular use case. Instead, add them in the components that need them.
```

Comment:
Question: what's the difference between this point and the one above?

Reply:
Oops, sorry bad copy paste when trying to incorporate AI suggestion 😆 deduped!